seed/seedwriter/writer_test.go

```diff
@@ -2111,17 +2119,42 @@ func (s *writerSuite) TestCore20NonDangerousNoChannelOverride(c *C) {
 				"type":            "gadget",
 				"default-channel": "20",
 			},
+			map[string]interface{}{
+				"name": "my-devmode",
+				"id":   s.AssertedSnapID("my-devmode"),
+				"type": "app",
+			},
 		},
 	})
 
-	s.opts.DefaultChannel = "stable"
 	s.opts.Label = "20191107"
+
+	const expectedErr = `cannot override channels, add local snaps or extra snaps with a model of grade higher than dangerous`
+
 	w, err := seedwriter.New(model, s.opts)
-	c.Assert(w, IsNil)
-	c.Check(err, ErrorMatches, `cannot override channels, add local snaps or extra snaps with a model of grade higher than dangerous`)
+	c.Assert(err, IsNil)
+
+	_, err = w.Start(s.db, s.newFetcher)
+	c.Assert(err, IsNil)
+
+	localSnaps, err := w.LocalSnaps()
+	c.Assert(err, IsNil)
+	c.Assert(localSnaps, HasLen, 0)
+
+	snaps, err := w.SnapsToDownload()
+	c.Check(err, IsNil)
+	c.Assert(snaps, HasLen, 5)
+
+	c.Assert(snaps[4].SnapName(), Equals, "my-devmode")
+	sn := snaps[4]
+
+	info := s.AssertedSnapInfo(sn.SnapName())
+	c.Assert(info, NotNil, Commentf("%s not defined", sn.SnapName()))
+	err = w.SetInfo(sn, info)
+	c.Assert(err, ErrorMatches, "cannot override channels, add devmode snaps, local snaps, or extra snaps with a model of grade higher than dangerous")
 }
 
-func (s *writerSuite) TestCore20NonDangerousDisallowedOptionsSnaps(c *C) {
+func (s *writerSuite) TestCore20NonDangerousNoChannelOverride(c *C) {
```

Comment:
did you merge two tests here? or one went away? bit confused by the diff here

Reply:
I don't know why the diff is so confusing but I just added one test I didn't modify the other one but git thinks I did something much more complicated

Reply:
no,  view file seems to agree that TestCore20NonDangerousDisallowedOptionsSnaps has disappeared

Reply:
sorry not sure how that happened, I have restore the test with a force push